src/pages/iou/IOUCurrencySelection.js

```diff
@@ -94,7 +95,7 @@ class IOUCurrencySelection extends Component {
      */
     getCurrencyOptions() {
         return _.map(this.props.currencyList, (currencyInfo, currencyCode) => {
-            const isSelectedCurrency = currencyCode === this.props.iou.selectedCurrencyCode;
+            const isSelectedCurrency = currencyCode === this.state.selectedCurrencyCode;
```

Comment:
The crash occurs here. This method`getCurrencyOptions` is called while initializing the state and trying to access the `state.selectedCurrencyCode` while it's not defined

Reply:
Ah, my bad. I didn't retest after moving the selected currency code to a state. So, I guess we have 2 options:

1. Take it out from the state and just put it back in a static variable
2. Update `getCurrencyOptions` to accept `selectedCurrencyCode`.

I personally prefer the first one as it's a static value. What do you think? @Santhosh-Sellavel @arosiclair

Reply:
After looking closer I think we should replace `selectedCurrencyCode` with a `getSelectedCurrencyCode` function instead. We can use that function in `getCurrencyOptions()` and `render()`. That should fix this while keeping everything reactive to prop changes.

Reply:
That's a nice idea. Updated and test well. Thanks for the help!